urns/phone_test.go

```diff
@@ -1,83 +1,77 @@
-package urns
+package urns_test
 
 import (
 	"testing"
 
 	"github.com/nyaruka/gocommon/i18n"
+	"github.com/nyaruka/gocommon/urns"
 	"github.com/stretchr/testify/assert"
 )
 
 func TestParsePhone(t *testing.T) {
 	testCases := []struct {
-		input    string
-		country  i18n.Country
-		expected URN
+		input       string
+		country     i18n.Country
+		expectedURN urns.URN
+		expectedErr string
 	}{
-		{" 0788383383 ", "RW", "tel:+250788383383"},
-		{"+250788383383 ", "RW", "tel:+250788383383"}, // already has country code and leading +
-		{"250788383383 ", "RW", "tel:+250788383383"},  // already has country code and no leading +
-		{"+250788383383 ", "KE", "tel:+250788383383"}, // already has a different country code
-		{"(917)992-5253", "US", "tel:+19179925253"},
-		{"800-CABBAGE", "US", "tel:+18002222243"},
-		{"+62877747666", "ID", "tel:+62877747666"},
-		{"0877747666", "ID", "tel:+62877747666"},
-		{"07531669965", "GB", "tel:+447531669965"},
-		{"263780821000", "ZW", "tel:+263780821000"},
-		{"254791541111", "US", "tel:+254791541111"}, // international but missing + and wrong country
+		{"+250788123123", "", "tel:+250788123123", ""},    // international number fine without country
+		{"+250 788 123-123", "", "tel:+250788123123", ""}, // still fine if not E164 formatted
+		{"250788123123", "", "tel:+250788123123", ""},     // still fine without leading + because it's long enough
 
-		{"1", "RW", "tel:1"},
-		{"123456", "RW", "tel:123456"},
-		{"mtn", "RW", "tel:mtn"},
-		{"!mtn!", "RW", "tel:mtn"}, // non tel chars stripped
+		{" 0788383383 ", "RW", "tel:+250788383383", ""},   // country code added
+		{"+250788383383 ", "RW", "tel:+250788383383", ""}, // already has country code and leading +
+		{"250788383383 ", "RW", "tel:+250788383383", ""},  // already has country code and no leading +
+		{"+250788383383 ", "KE", "tel:+250788383383", ""}, // already has a different country code
+		{"(917)992-5253", "US", "tel:+19179925253", ""},   // punctuation removed
+		{"800-CABBAGE", "US", "tel:+18002222243", ""},     // vanity numbers converted to digits
+		{"+62877747666", "ID", "tel:+62877747666", ""},
+		{"0877747666", "ID", "tel:+62877747666", ""},
+		{"07531669965", "GB", "tel:+447531669965", ""},
+		{"263780821000", "ZW", "tel:+263780821000", ""},
+		{"254791541111", "US", "tel:+254791541111", ""}, // international but missing + and wrong country
 
-		{"0788383383", "ZZ", NilURN}, // invalid country code
-		{"1234567890123456789012345678901234567890123456789012345678901234567890123456789", "RW", NilURN}, // too long
-	}
+		{"1234", "US", "tel:1234", ""},
+		{"12345", "US", "tel:12345", ""},
+		{"123", "RW", "tel:123", ""},
 
-	for i, tc := range testCases {
-		urn, err := ParsePhone(tc.input, tc.country)
+		{"1", "RW", "", "the phone number supplied is not a number"},
+		{"1234", "RW", "", "not a possible number or shortcode"},     // RW short codes are 3 digits
```

Comment:
@norkans7 true ?

Reply:
wrong, there are short code with 4 digits

Reply:
are those dialable numbers ?

Reply:
I notice here in Ecuador I get messages from numbers like 8080 which libphonenumber doesn't recognize but I thought maybe they're only one way

Reply:
The alphanumeric are the one that are one way

I think the rule for short code is length shorte than or equal to 6 digits

Reply:
I guess we could just say any 1-5 digits is a short code ?

Reply:
1-6 digits I think

Reply:
Reading through https://en.wikipedia.org/wiki/Short_code and it seems some countries allow 7 or 8 digits.. which then overlaps with local numbers. So maybe we say it's valid *if* libphonenumber says it's valid, *or* it's 2-4 digits.

It's probably ok if there's a short code we reject here and there. You generally don't want to be talking to short codes because they are generally connected to other automated systems.

Reply:
@norkans7 updated to accept short codes that libphonenumber recognizes OR anything that matches `^[1-9][0-9]{2,4}$`, e.g. 3-5 digits, not starting with a zero

Reply:
why not 3-6 digits?

Reply:
I'm assuming that libphonenumber mostly recognizes valid short codes so this is just a fallback.. and I'm trying to be careful to not start accepting local numbers as valid.. but we can make it 6 I guess.... updating...

Reply:
I'd like to move all phone number validation into go eventually so that logic is consistent and in one place.